interface/drugs/drugs.inc.php

```diff
@@ -74,6 +74,17 @@ function sellDrug(
 
   // error_log("quantity = '$quantity'"); // debugging
 
+    // Sanity check.
+    if (!$testonly) {
+        $tmp = sqlQuery(
+            "SELECT count(*) AS count from form_encounter WHERE pid = ? AND encounter = ?",
+            array($patient_id, $encounter_id)
+        );
+        if (empty($tmp['count'])) {
+            die(xlt('Internal error: the referenced encounter no longer exists.') . " $patient_id $encounter_id");
```

Comment:
text() around $patient_id and $encounter_id

Reply:
Done.